src/backend/distributed/commands/type.c

```diff
@@ -493,7 +494,6 @@ PlanDropTypeStmt(DropStmt *stmt, const char *queryString)
 	/*
 	 * remove the entries for the distributed objects on dropping
 	 */
-	distributedTypeAddresses = TypeNameListToObjectAddresses(distributedTypes);
 	foreach(addressCell, distributedTypeAddresses)
 	{
 		ObjectAddress *address = (ObjectAddress *) lfirst(addressCell);
```

Comment:
Maybe we could iterate over `distributedTypes` and use a `TypeNameToObjectAddress` function here and in `FilterNameListForDistributedTypes`? Seems cleaner overall and the double work is easily forgiven in a DROP TYPE statement.

Reply:
Potentially, still means translating name-to-address twice when copying this code to a new object type